### PR TITLE
feat: jenkins file for qualifying dev branch

### PR DESCRIPTION
**Pull request by cdurai-netapp**
1. update existing jenkins file to pick branch, docker and rpm build
2. create a single workflow jenkins file to quality dev branch for every commit

### CODE REVIEW

File: jenkins/artifacts/JenkinWorkflow [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+               apt-get install -y build-essential
+                '''
+            }
+        }
+
+        stage('Git Clone Harvest') {
+            steps {
+               sh '''
+                #git clone --single-branch --branch $GIT_BRANCH https://github.com/NetApp/harvest.git
+                git clone --single-branch --branch main https://github.com/NetApp/harvest.git
```
Comment (rahulguptajss) [inline jenkins/artifacts/JenkinWorkflow:46]: may be rename the jenkins file if branch name is going to main only.

File: jenkins/artifacts/JenkinWorkflow [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    archiveArtifacts artifacts: '**', fingerprint: true
+                }
+            }
+        }
+        stage('Run Tests') {
+            steps {
+                script {
+                    dockerBuild = "${BUILD_URL}/artifact/docker_harvest.tar"
+                    nativeBuild = "${BUILD_URL}/artifact/harvest.tar.gz"
+                    build job: 'harvest2_0/smoke', parameters: [string(name: 'BRANCH', value: "${GIT_BRANCH}"), string(name: 'NATIVE', value: "${nativeBuild}"), string(name: 'DOCKER', value: "${dockerBuild}")]
```
Comment (rahulguptajss) [inline jenkins/artifacts/JenkinWorkflow:90]: GIT_BRANCH is not defined in this file

**Review by rahulguptajss — CHANGES_REQUESTED** [review]

File: jenkins/artifacts/JenkinWorkflow (hunk 1 above)
Comment (cdurai-netapp) [reply jenkins/artifacts/JenkinWorkflow:46]: for testing purpose i have kept main. will revert it to branch name

File: jenkins/artifacts/JenkinWorkflow (hunk 2 above)
Comment (cdurai-netapp) [reply jenkins/artifacts/JenkinWorkflow:90]: GIT_BRANCH is an env variable hence didnt defined in the jenkins file.

Comment (rahulguptajss) [reply jenkins/artifacts/JenkinWorkflow:90]: shouldn't env variable also be in jenkinsfile? Do you mean to set these variables in jenkins setup?

Comment (cdurai-netapp) [reply jenkins/artifacts/JenkinWorkflow:90]: these are in built env variable from git plugin

https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/Complete-Jenkins-Git-environment-variables-list-for-batch-jobs-and-shell-script-builds

**Review by cgrinds — APPROVED** [review]

**Review by rahulguptajss — APPROVED** [review]

**Pull request merged**